plugins/inputs/amqp_consumer/amqp_consumer.go

```diff
@@ -115,7 +115,7 @@ func (a *AMQPConsumer) Init() error {
 	}
 
 	if a.MaxDecompressionSize <= 0 {
-		a.MaxDecompressionSize = internal.DefaultMaxDecompressionSize
+		a.MaxDecompressionSize = config.Size(internal.DefaultMaxDecompressionSize)
 	}
```

Comment:
Please change to

```golang
	var options []internal.DecodingOption
	if a.MaxDecompressionSize > 0 {
		options = append(options, internal.DecoderMaxDecompressionSize(int64(a.MaxDecompressionSize)))
	}
```

and then use these options in like 166 like
```golang
	a.decoder, err = internal.NewContentDecoder(a.ContentEncoding, options...)
```

Reply:
but these are two different functions.

Reply:
I think it's better for you to do the refactor around these plugins. I'm not sure if I'll break something.